### PR TITLE
git checkout: add example for creating new branch based on remote branch with different name

### PR DESCRIPTION
**Pull request by mikerogne**
Adds an example for creating and checking out a new branch that's based on an existing remote/source branch with a different name.

It's similar to https://github.com/tldr-pages/tldr/pull/1359 - but shows the example of allowing the new branch being created having a different branch name than the source.

It's something I don't use super often, but end up having to Google each time because it's not in `tldr git checkout` :)

----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited.

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

File: pages/common/git-checkout.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 `git checkout -b {{branch_name}}`
 
+- Create and switch to a new branch based on an existing remote branch:
+
+`git checkout -b {{branch_name}} {{remote/source_branch}}`
```
Comment (mfrw) [inline pages/common/git-checkout.md]: Usually when we checkout a remote branch, we also want to track it. Should we add that option too or will it be too complicated? @agnivade @sbrl @pxgamer 
```suggestion
`git checkout -b {{branch_name}} {{remote/source_branch}} --track`
```

Comment (mikerogne) [reply pages/common/git-checkout.md]: Heyas! Thanks for checking it out so fast. No pun intended 👈👈😎.

My use-case for it was to demonstrate creating a new branch based on a source branch that has a different name, even from a remote. There was some --track/--tracking option mentioned in the other PR I referenced in the description. Not sure. Up to you guys!

**Comment by mikerogne** [comment]
I think https://github.com/tldr-pages/tldr/pull/2494/commits/c2c7434f645a3702289c97da55ad34c150b1388c is the clearest description I can come up with. Valid examples are included. Tried my best to keep it very succinct.

**Comment by FatBoyXPC** [comment]
Pretty slick change, I used to have to google this quite often!

**Pull request merged**